CenterForOpenScience/osf.io
# [ENG-2133] Admin App for Registries moderation

## Purpose

Add functionality for admin app ahead of registries moderation project.

## Changes

- adds ability to add/remove moderators from provider
- adds a few new fields to edit provider such as moderation type and comment visibility

![Screenshot_2020-09-29 Registration Provider(1)](https://user-images.githubusercontent.com/9688518/94607104-7accdd00-0269-11eb-8188-037daaf1cd8a.png)

<img width="1438" alt="Screen Shot 2020-09-30 at 1 04 44 PM" src="https://user-images.githubusercontent.com/9688518/94716869-8679db80-031d-11eb-9421-edbe00c0daa1.png">

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify all links go to logically where you would think
- Verify all text inputs are validated and behave correctly
- Verify that everything necesarry is editable on the registration provider.
- Verify moderators can be added/removed correctly
- Verify non-functional requirements are met, if it's anything needs bulk updates etc? 

What are the areas of risk?

Permissions are the greatest risk here, but everything should be acceptable for admins to do, so nothing complicated.

Any concerns/considerations/questions that development raised?

Some scope questions about how much on the registration should be editable.

## Documentation

admin stuff is institutional knowledge! passed down from generation to generation!

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2133
